compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3971,6 +3971,11 @@ private void checkFunctionInvocationExpr(BLangInvocation iExpr) {
             // This will enable, type param support, if the function is called directly.
             this.env = SymbolEnv.createInvocationEnv(iExpr, this.env);
         }
+        // Log an error if the function returns never and invoked invalidly.
+        if (funcSymbol.type.getReturnType().tag == TypeTags.NEVER &&
+                env.node.getKind() != NodeKind.EXPRESSION_STATEMENT) {
```

Comment:
Should this be in the code analyzer? Not exactly a type checking logic right?

Reply:
Will change this